### PR TITLE
Refactor signup with hooks #7

### PR DESCRIPTION
**Pull request by palmdaldaniel**
Pls test and see what you think

### CODE REVIEW

**Comment by mikhael28** [comment]
Hey @palmdaldaniel thank you for sending this in! The hook is functional and things work, but I would like you to resolve a couple of things before merging!

File: src/profile/Signup.js
Comment (mikhael28) [inline src/profile/Signup.js:19]: Can you remove these comments and the TODO at the top? Once the issue has been resolved, there is no need to keep this here :)

File: src/profile/Signup.js
Comment (mikhael28) [inline src/profile/Signup.js:70]: Instead of this alert, can you please use the `errorToast` function, that is exported inside `src/libs/toast`?

File: src/profile/Signup.js
Comment (mikhael28) [inline src/profile/Signup.js:67]: After Auth.signIn, can you run the `successToast` function that is also exported from the libs toast file? We want to give our users some sort of visual proof that their request has either succeeded, or failed.

File: src/profile/Signup.js
Comment (mikhael28) [inline src/profile/Signup.js:83]: Can you please install Prettier on VS Code (assuming you are using VS Code), and can you run the linting command (auto-formatting, makes the code look pretty) as the very last commit? Why the very last? Because the commit will modify 50% of the file, and will make it hard for me to see the other changes. I have Prettier set to run every time I save the file, command+s. It keeps the code looking pretty, and easy for maintainers like myself to see what the changes are.

**Review by mikhael28 — CHANGES_REQUESTED** [review]
Please make these requested changes - thank you for sending this in.

File: src/profile/Signup.js
Comment (palmdaldaniel) [reply src/profile/Signup.js:70]: same message ? :)

File: src/profile/Signup.js
Comment (palmdaldaniel) [reply src/profile/Signup.js:83]: alright,  will format it with prettier

File: src/profile/Signup.js
Comment (palmdaldaniel) [reply src/profile/Signup.js:67]: Ok, what param do you want me to feed it with ?

File: src/profile/Signup.js
Comment (mikhael28) [reply src/profile/Signup.js:70]: Take a look at how error toast function is used in other parts of the repository. In general, you pass it the `error` e object in the catch block.

**Comment by palmdaldaniel** [comment]
Have implemented the changes now. Have a look pls 😀

**Comment by mikhael28** [comment]
Hey @palmdaldaniel great stuff - I'm going to merge your code in! Congratulations - when you have a chance, can you please introduce yourself at https://github.com/mikhael28/paretOS/discussions/17 ?

As an aside, you mentioned that you don't have any more emails to test signup with. I will teach you a secret - if you have a Gmail account, you can register mygmail+somestring@gmail.com - the "+somestring" is an alias, and any email that gets sent to that email address will automatically be sent to the inbox of mygmail@gmail.com - but, all the login services and such will treat it as a unique email. That's how I do my testing - I can use michael+jamesbond@pareto.education and create a new dev account instantly. Cheers, make sure to introduce yourself!

**Pull request merged**
